test/Datadog.Trace.DuckTyping.Tests/ObscureObject.cs

```diff
@@ -0,0 +1,976 @@
+
+            private string PrivateGetReferenceType { get; } = "33";
+
+            // *
+
+            public string PublicGetSetReferenceType { get; set; } = "40";
+
+            internal string InternalGetSetReferenceType { get; set; } = "41";
+
+            protected string ProtectedGetSetReferenceType { get; set; } = "42";
+
+            private string PrivateGetSetReferenceType { get; set; } = "43";
+
+            // *********
+
+            public static DummyFieldObject PublicStaticGetSelfType { get; } = DummyFieldObject.Default;
+
+            internal static DummyFieldObject InternalStaticGetSelfType { get; } = DummyFieldObject.Default;
+
+            protected static DummyFieldObject ProtectedStaticGetSelfType { get; } = DummyFieldObject.Default;
+
+            private static DummyFieldObject PrivateStaticGetSelfType { get; } = DummyFieldObject.Default;
+
+            // *
+
+            public static DummyFieldObject PublicStaticGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            internal static DummyFieldObject InternalStaticGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            protected static DummyFieldObject ProtectedStaticGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            private static DummyFieldObject PrivateStaticGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            // *
+
+            public DummyFieldObject PublicGetSelfType { get; } = DummyFieldObject.Default;
+
+            internal DummyFieldObject InternalGetSelfType { get; } = DummyFieldObject.Default;
+
+            protected DummyFieldObject ProtectedGetSelfType { get; } = DummyFieldObject.Default;
+
+            private DummyFieldObject PrivateGetSelfType { get; } = DummyFieldObject.Default;
+
+            // *
+
+            public DummyFieldObject PublicGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            internal DummyFieldObject InternalGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            protected DummyFieldObject ProtectedGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            private DummyFieldObject PrivateGetSetSelfType { get; set; } = DummyFieldObject.Default;
+
+            // ***
+
+            public static int? PublicStaticNullableInt { get; set; }
+
+            private static int? PrivateStaticNullableInt { get; set; }
+
+            public int? PublicNullableInt { get; set; }
+
+            private int? PrivateNullableInt { get; set; }
+
+            // ***
+
+            public int this[int index]
+            {
+                get => _dictioInt[index];
+                set => _dictioInt[index] = value;
+            }
+
+            public string this[string index]
+            {
+                get => _dictioString[index];
+                set => _dictioString[index] = value;
+            }
+
+            // ***
+
+            public TaskStatus Status { get; set; } = TaskStatus.RanToCompletion;
```

Comment:
I don't think this got used in the Property tests, are you missing a couple of test cases?

Reply:
Added a test case for this in https://github.com/DataDog/dd-trace-dotnet/pull/946/commits/e3020964dcbb2b80b94368777abbdd5b1de4e248